### PR TITLE
Interdiff no longer reversed.

### LINKED ISSUE #47: Interdiffs seem to be reversed
Using `drupalorg issue:interdiff 3091587` I got an interdiff, but it was reversed.

```
diff --git a/src/Form/SubscriptionCancelConfirmForm.php b/src/Form/SubscriptionCancelConfirmForm.php
index 316abb1..804920d 100644
--- a/src/Form/SubscriptionCancelConfirmForm.php
+++ b/src/Form/SubscriptionCancelConfirmForm.php
@@ -27,12 +27,11 @@ class SubscriptionCancelConfirmForm extends EntityConfirmFormBase {
 
     $form['scheduled'] = [
       '#type' => 'radios',
-      '#title' => $this->t('Cancellation date'),
+      '#title' => 'When to cancel?',
+      '#description' => "A subscription can be cancelled immediately, or scheduled to be canceled at the end of the subscription's billing period.",
       '#options' => [
-        'scheduled' => $this->t('End of the current billing period (:period)', [
-          ':period' => $subscription->getCurrentBillingPeriod()->getEndDate()->format('M jS Y'),
-        ]),
-        'now' => $this->t('Immediately'),
+        'scheduled' => 'Schedule cancellation at end of billing period.',
+        'now' => 'Cancel subscription immediately.',
       ],
       '#default_value' => 'scheduled',
     ];
@@ -48,14 +47,16 @@ class SubscriptionCancelConfirmForm extends EntityConfirmFormBase {
     $subscription = $this->entity;
     assert($subscription instanceof SubscriptionInterface);
 
-    $subscription->cancel($scheduled);
-    $subscription->save();
-
-    if ($scheduled) {
-      $this->messenger()->addMessage(t('This subscription has been scheduled for cancellation.'));
+    if ($subscription->cancel($scheduled)->save()) {
+      if ($scheduled) {
+        $this->messenger()->addMessage(t('This subscription has been scheduled for cancellation.'));
+      }
+      else  {
+        $this->messenger()->addMessage(t('This subscription has been cancelled.'));
+      }
     }
     else {
-      $this->messenger()->addMessage(t('This subscription has been cancelled.'));
+      $this->messenger()->addMessage(t('There was a problem with cancelling the subscription.'), 'error');
     }
     $form_state->setRedirect('entity.commerce_subscription.edit_form', ['commerce_subscription' => $subscription->id()]);
   }
```

### PR DESCRIPTION
**Pull request by hansfn**
Also added some debugging and a sanity check.

Fixes #47.

### CODE REVIEW

**Comment by hansfn** [comment]
I selected to use the two last commits in the local issue branch to create the interdiff. That is useful when you are working on the same local branch for a while - without recreating it. (Typically when you are the only one working on the issue.)

I imagine that someone would like to compare the first and last commit if they have made changes in separate commits. If so, they can rebase the local branch. We are adding one patch to the drupal issue queue anyway.

**Comment by mglaman** [comment]
> That is useful when you are working on the same local branch for a while - without recreating it. (Typically when you are the only one working on the issue.)

This should still work, too, if you're applying other folk's interdiff changes. And yeah, if things get out of whack it's probably a good idea to reset the branch.

File: src/Cli/Command/Issue/Interdiff.php [hunk 1]
```diff
@@ -67,18 +67,29 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
     }
 
-    // Create a diff from the first commit of the issue branch.
-    $first_issue_branch_commit = sprintf('$(git log %s..HEAD --format=%%H)', $issue_version_branch);
-    $process = new Process(sprintf('git diff %s', $first_issue_branch_commit));
+    // Find the two last commits on the issue branch.
+    $process = new Process(sprintf('git log -2 %s..HEAD --format=%%H', $issue_version_branch));
     $process->run();
+    $last_issue_branch_commits = explode(PHP_EOL, $process->getOutput());
+    array_pop($last_issue_branch_commits); // Empty item because line breaks from "git log" output.
```
Comment (mglaman) [inline src/Cli/Command/Issue/Interdiff.php:78]: Would it be better to array_filter and strip out `PHP_EOL` and use `trim`? just to be safe.

File: src/Cli/Command/Issue/Interdiff.php [hunk 2]
```diff
@@ -67,18 +67,29 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
+    // Create a diff between two last commits of the issue branch. (Reverse order of output from "git log".)
+    $diff_cmd = sprintf('git diff %s', implode(" ", array_reverse($last_issue_branch_commits)));
+    $process = new Process($diff_cmd);
+    $process->run();
     $filename = $this->cwd . DIRECTORY_SEPARATOR . $this->buildInterdiffName($issue);
     file_put_contents($filename, $process->getOutput());
     $this->stdOut->writeln("<comment>Interdiff written to {$filename}</comment>");
 
-    $process = new Process(sprintf('git diff %s --stat', $first_issue_branch_commit));
+    $process = new Process("$diff_cmd --stat");
```
Comment (mglaman) [inline src/Cli/Command/Issue/Interdiff.php:92]: Looks good, I tested the commands locally and it properly does the diff.

File: src/Cli/Command/Issue/Interdiff.php (hunk 1 above)
Comment (hansfn) [reply src/Cli/Command/Issue/Interdiff.php]: Yes, that is much better / safer. array_filter is one of those function I tend to forget.

File: src/Cli/Command/Issue/Interdiff.php (hunk 2 above)
Comment (hansfn) [reply src/Cli/Command/Issue/Interdiff.php:92]: I'm happy it worked for you too. Thanks for merging so quickly.

**Pull request merged**